### DISCUSSION TITLE
HMS1600 - TX SPI TIMEOUT

### DISCUSSION THREAD

**Discussion by pi-amp** [tbnobody/OpenDTU#2433] [Q&A]
Hi,
I have set up an opendtu for 6 monthes with an esp32 and CMT2300A daugther board, and a HMS1600 inverter.
It worked perfectly, but began to misbehave 2 weeks ago.
About every day, i loose connection with the inverter and the console repeateadly says "TX SPI TIMEOUT".
A software reboot of the opendtu, or a unplug/plug of the power, allows the connection to come back, but it goes away after one day.

What could be wrong ? 
thanks!

**Comment by robsy** [comment]
Hi,

it seems that I have a similar problem. Can you try to change the TX power next time? At my setup it's enough to restart the connection.

https://github.com/tbnobody/OpenDTU/issues/2438

Robert

**Comment by pi-amp** [comment]
I gave some air to the cmt module by putting in free on the dtu box, and it's been working fine since some days. Maybe the module was overheating ?
